src/test/function_test/test_copy.cpp

```diff
@@ -0,0 +1,257 @@
+        c = CCH[random() % sizeof(CCH)];
+    }
+
+    std::string hash_key;
+    std::string sort_key;
+    std::string value;
+    while (base_data[empty_hash_key].size() < 1000) {
+        sort_key = random_string();
+        value = random_string();
+        int ret = srouce_client->set(empty_hash_key, sort_key, value);
+        ASSERT_EQ(PERR_OK, ret) << "Error occurred when set, hash_key=" << hash_key
+                                << ", sort_key=" << sort_key
+                                << ", error=" << srouce_client->get_error_string(ret);
+        base_data[empty_hash_key][sort_key] = value;
+    }
+
+    while (base_data.size() < 500) {
+        hash_key = random_string();
+        while (base_data[hash_key].size() < 10) {
+            sort_key = random_string();
+            value = random_string();
+            int ret = srouce_client->set(hash_key, sort_key, value);
+            ASSERT_EQ(PERR_OK, ret) << "Error occurred when set, hash_key=" << hash_key
+                                    << ", sort_key=" << sort_key
+                                    << ", error=" << srouce_client->get_error_string(ret);
+            base_data[hash_key][sort_key] = value;
+        }
+    }
+
+    ddebug("Data filled.");
+}
+
+class copy_data_test : public testing::Test
+{
+public:
+    static void SetUpTestCase()
+    {
+        ddebug("SetUp...");
+        create_table_and_get_client();
+        fill_data();
+    }
+
+    static void TearDownTestCase()
+    {
+        ddebug("TearDown...");
+        chdir(global_env::instance()._pegasus_root.c_str());
+        system("./run.sh clear_onebox");
+        system("./run.sh start_onebox -w");
+        chdir(global_env::instance()._working_dir.c_str());
+    }
+};
+
+TEST_F(copy_data_test, EMPTY_HASH_KEY_COPOY)
+{
+    ddebug("TESTING_COPY_DATA, EMPTY HAS_HKEY COPOY ....");
+
+    pegasus_client::scan_options options;
+    options.return_expire_ts = true;
+    std::vector<pegasus::pegasus_client::pegasus_scanner *> raw_scanners;
+    int ret = srouce_client->get_unordered_scanners(INT_MAX, options, raw_scanners);
+    ASSERT_EQ(pegasus::PERR_OK, ret) << "Error occurred when getting scanner. error="
+                                     << srouce_client->get_error_string(ret);
+
+    ddebug("INFO: open source app scanner succeed, partition_count = %d\n",
+           (int)raw_scanners.size());
+
+    std::vector<pegasus::pegasus_client::pegasus_scanner_wrapper> scanners;
+    for (auto raw_scanner : raw_scanners) {
+        ASSERT_NE(nullptr, raw_scanner);
+        scanners.push_back(raw_scanner->get_smart_wrapper());
+    }
+    raw_scanners.clear();
+
+    int split_count = scanners.size();
+    ddebug("INFO: prepare scanners succeed, split_count = %d\n", split_count);
+
+    std::atomic_bool error_occurred(false);
+    std::vector<std::unique_ptr<scan_data_context>> contexts;
+    std::unique_ptr<pegasus::geo::geo_client> geo_client;
+
+    for (int i = 0; i < split_count; i++) {
```

Comment:
you should run the copy data shell tool, not the function, otherwise the tool is not coveraged by you test.